From 0dbf8ab47534d50c33f2a84666313ac0023a710d Mon Sep 17 00:00:00 2001
From: Richard Laager <rlaager@wiktel.com>
Date: Sat, 10 Dec 2016 23:19:46 -0600
Subject: [PATCH 2/5] ntpviz: Fix temp file binary vs. text

This fixes the following error from os.write() on Python 3:
TypeError: a bytes-like object is required, not 'str'
---
 ntpclients/ntpviz | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/ntpclients/ntpviz b/ntpclients/ntpviz
index a564e7b..828fee5 100755
--- a/ntpclients/ntpviz
+++ b/ntpclients/ntpviz
@@ -290,14 +290,14 @@ def gnuplot(template, outfile=None):
 
     # can be 30% faster to write to a tmp file than to pipe to gnuplot
     # bonus, we can keep the plot file for debug.
-    tmp_file, tmp_filename = tempfile.mkstemp( suffix='.plt')
+    tmp_file = tempfile.NamedTemporaryFile(mode='w', suffix='.plt', delete=False)
     # note that tmp_file is a file handle, it is not a file object
-    os.write( tmp_file, template)
-    os.close(tmp_file)
+    tmp_file.write(template)
+    tmp_file.close()
 
     # shell=True is a security hazard, do not use
     try:
-        rcode = subprocess.call( ['gnuplot', tmp_filename], stdout=out)
+        rcode = subprocess.call(['gnuplot', tmp_file.name], stdout=out)
     except OSError as e:
         if e.errno == os.errno.ENOENT:
             # gnuplot not found
@@ -309,12 +309,12 @@ def gnuplot(template, outfile=None):
 
     if 0 != rcode:
         sys.stderr.write("ntpviz: WARNING: plot returned %s\n" % rcode)
-        sys.stderr.write("ntpviz: WARNING: plot file %s\n" % tmp_filename)
+        sys.stderr.write("ntpviz: WARNING: plot file %s\n" % tmp_file.name)
     elif 2 <= args.debug_level:
-        sys.stderr.write("ntpviz: INFO: plot file %s\n" % tmp_filename)
+        sys.stderr.write("ntpviz: INFO: plot file %s\n" % tmp_file.name)
     else:
         # remove tmp file
-        os.remove(tmp_filename)
+        os.remove(tmp_file.name)
 
     return rcode
 
-- 
2.7.4

